src/lib/protocols/ssh.c

```diff
@@ -54,23 +59,46 @@
   that usually is packet 14
 */
 
-/* #define SSH_DEBUG 1 */
+#define SSH_DEBUG 1
 
 static void ndpi_search_ssh_tcp(struct ndpi_detection_module_struct *ndpi_struct, struct ndpi_flow_struct *flow);
-  
+
 /* ************************************************************************ */
 
-static void ssh_analyse_signature_version(struct ndpi_detection_module_struct *ndpi_struct,
+static void ssh_analyze_signature_version(struct ndpi_detection_module_struct *ndpi_struct,
 					  struct ndpi_flow_struct *flow,
 					  char *str_to_check,
 					  u_int8_t is_client_signature) {
 
+  if (str_to_check == NULL) return;
+
+  int major = 0;
+  int minor = 0;
+  int patch = 0;
+  int obsolete_ssh_version = 0;
+
+  if (sscanf(str_to_check, "SSH-%*f-OpenSSH_%d.%d.%d", &major, &minor, &patch) < 0) 
```

Comment:
IMHO that should be more generalized as there are **lots** of other SSH implementation available besides `OpenSSH`.

Reply:
if(sscanf(....)  != 3)...
Do no use generic checks <0, >0 etc